### PR TITLE
FIX: Ship saved search allowed attribute from server side

### PR DESCRIPTION
**Pull request by tgxworld**
Modifying the user class on the client side is brittle as the current user
is initialised before the initializers run. When that happens, the computed property that is added by the plugin will not be defined and thus result in the saved search link not displaying as we rely on the computed property to determine if the link should be displayed.

### CODE REVIEW

**Review by martin-brennan — APPROVED** [review]
Nice, this is way better :rocket:

File: spec/serializers/user_serializer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      serialized = UserSerializer.new(user, scope: Guardian.new(user), root: false).as_json
+
+      expect(serialized[:author_level]).to eq(0)
+      expect(serialized[:saved_searches_allowed]).to eq(false)
+    end
+
+    it 'serializes to true when user is not staff but has required trust level and author level' do
+      SiteSetting.saved_searches_min_trust_level = TrustLevel[2]
+      user.update!(trust_level: TrustLevel[2])
+      badge = Fabricate(:badge, name: "author level #{SavedSearches::REQUIRED_AUTHOR_LEVEL + 1}")
```
Comment (nattsw) [inline spec/serializers/user_serializer_spec.rb:62]: Btw, why did you add 1 here?

File: plugin.rb [hunk 2]
```diff
@@ -28,19 +28,35 @@
...
+    object.trust_level >= SiteSetting.saved_searches_min_trust_level &&
+      self.author_level > SavedSearches::REQUIRED_AUTHOR_LEVEL
+  end
+
+  add_to_serializer(:user, :include_saved_searches_allowed?, false) do
+    plugin.enabled? && scope.can_edit?(object)
+  end
+
   module ::SavedSearches
+    REQUIRED_AUTHOR_LEVEL = 2
```
Comment (nattsw) [inline plugin.rb:58]: Is `author_level` something specific to the plugin?

Comment (nattsw) [reply plugin.rb:58]: Hmm nevermind I see it's from the deleted JS file.

**Review by nattsw — APPROVED** [review]

**Comment by tgxworld** [comment]
Thank you for the review @nattsw and @martin-brennan :heart:
